src/transformers/trainer.py

```diff
@@ -3518,11 +3519,16 @@ def log(self, logs: Dict[str, float]) -> None:
         Args:
             logs (`Dict[str, float]`):
                 The values to log.
+            start_time (`Optional[float]`):
+                The start of training.
         """
         if self.state.epoch is not None:
             logs["epoch"] = self.state.epoch
         if self.args.include_num_input_tokens_seen:
             logs["num_input_tokens_seen"] = self.state.num_input_tokens_seen
+            if start_time is not None:
+                runtime = time.time() - start_time
+                logs["tokens_per_second"] = round(self.state.num_input_tokens_seen / runtime, 3)
```

Comment:
Maybe use `speed_metrics` here and we can also update the description of `include_num_input_tokens_seen` since we also including the speed metrics for tokens per second

Reply:
Corrected and rebased, thanks!